### ISSUE TITLE
sync via syncthing

### ISSUE THREAD

**Issue by RustoMCSpit** [quillpad/quillpad#219]
nextcloud shouldnt be the only method

**Comment by sigmonsays** [comment]
I found this issue looking for a solution to use syncthing with quillpad.

It is my understanding that android makes it hard to access the storage in consistent ways. However, i know it can be done as other apps do it.

**Comment by englut** [comment]
I do not want to install Nextcloud and would much prefer a filesystem sync (sync things fork uses this) if that isn't too difficult to achieve!

**Comment by ndgnuh** [comment]
This is currently available because we can export to the file system now. But it is buggy because:

- The restored notes are duplicated
- The notes are not put to their respective notebooks

(I'm running Quillpad 1.5.1)

**Comment by jsixface** [comment]
Syncing via syncthing specifically not planned. As @ndgnuh mentioned, you can export the notes to the file storage.

**Comment by sigmonsays** [comment]
I disagree with this decision since exporting is a manual step and sync is supposed to be manual. 

Good luck!

**Comment by englut** [comment]
Yeah... unless the exporting can happen automatically, it's not really a great solution.

**Comment by ndgnuh** [comment]
@englut @sigmonsays 

I said "export" in my previous reply, which is a little misleading. It is actually automatic, in the settings, under "Syncing service" select "File Storage".


@jsixface 
However, I think the duplication bug should really be fixed. It is really annoying to have duplicate notes with (1), (2), (3)... suffixes. It also messes up which note is the latest one (if I edit them on multiple devices) and the note is not kept in their respective notebook when restored.

Maybe storing the note on the file system using UUID's as the file's name should help?

### Update 1

~~Sorry for the noise, it is actually not a QuillPad's bug (at least on my end) at all.~~ After clearing the app's data, the app still restores duplicates. So I went and find where they were restored from. 

Turns out, my Syncthing was having file versioning enabled. It keeps the previous versions of the notes in `.stversion` folder. When QuillPad sync *from* the file system, it also restores the note in the version-control directory. And when QuillPad sync *to* the file system, it also syncs the duplicates along with the regular ones, creating a cycle of duplications.

TL;DR: Turn off Syncthing versioning on QuillPad's sync folder, or having it store the version somewhere else.

My suggestion on using UUID remains though, the display name of the notes seem too "fragile" to be used as an identifier.

### Update 2

It is actually both, I deleted the app data and restore the note from file storage, Quillpad still displays duplicates.

**Comment by jsixface** [comment]
@ndgnuh Thanks for the insights, indeed the backups/versions under the hidden folder makes sense. 

>  I deleted the app data and restore the note from file storage, Quillpad still displays duplicates.

Did you mean, you deleted the app and made sure there are no notes under any hidden directories in file storage and yet Quillpad is creating duplicates?

**Comment by ndgnuh** [comment]
> [@ndgnuh](https://github.com/ndgnuh) Thanks for the insights, indeed the backups/versions under the hidden folder makes sense.
> 
> > I deleted the app data and restore the note from file storage, Quillpad still displays duplicates.
> 
> Did you mean, you deleted the app and made sure there are no notes under any hidden directories in file storage and yet Quillpad is creating duplicates?

Yes, I deleted all the duplicates, clear QuillPad's data, go into the app and set the synchronization directory. All the notes are imported, but they appear twice (No `(1)` suffixes). The duplicates do not show up in the sync directory, but when exporting (using the backup option) the duplicate notes appeared.
